### PR TITLE
systemd-journal: fix --with-systemd-journal=optional mode

### PR DESCRIPTION
**Pull request by HofiOne**
Further modifications in the lookup of sd_id128_XXX symbols (that can be presented in separate modules on some OSes e.g. Centos)

Signed-off-by: Hofi <hofione@gmail.com>

### CODE REVIEW

File: modules/systemd-journal/journald-subsystem.c [hunk 1]
```diff
@@ -57,20 +57,25 @@ int (*sd_id128_get_boot)(sd_id128_t *ret);
 
 #define JOURNAL_LIBRARY_NAME "libsystemd-journal.so.0"
 #define SYSTEMD_LIBRARY_NAME "libsystemd.so.0"
+#define ID128_LIBRARY_NAME   "libsystemd-id128.so"
 
 const char *journald_libraries[] = {JOURNAL_LIBRARY_NAME, SYSTEMD_LIBRARY_NAME, NULL};
+const char *journald_ext_libraries[] = {ID128_LIBRARY_NAME, JOURNAL_LIBRARY_NAME, SYSTEMD_LIBRARY_NAME, NULL};
 
 static GModule *journald_module;
+static GModule *journald_ext_module;
```
Comment (bazsi) [inline modules/systemd-journal/journald-subsystem.c:66]: we could probably move these to local variables in load_journald_subsystem() and then pass them along
using arguments to _load_journald_subsystem()

**Review by bazsi — APPROVED** [review]
I am fine with this as is, I added a small improvement that would have made reviewing this easier but that's optional too.

With that change, this unit would be drastically more readable than it has been a month ago :)

Thanks

File: modules/systemd-journal/journald-subsystem.c (hunk 1 above)
Comment (HofiOne) [reply modules/systemd-journal/journald-subsystem.c:66]: yeah, it came to my mind as well, the only reason I left it this way is that once we might want to add a really proper cleanup then we might want to call unload_journald_subsystem() and that would not be that nice to pass these to that function too

**Comment by HofiOne** [comment]
@bazsi Sorry, I've made a typo in libsystemd-id128.so.0, corrected, and added some logging of the lookup process and possible errors. Hope it's fine now.

**Comment by HofiOne** [comment]
Centos tests are passed, seems we are fine now

**Comment by HofiOne** [comment]
@kira-syslogng retest this please;

**Comment by HofiOne** [comment]
@bazsi Could you please take a look at this one more (and hopefully the last time) again?
Thanks!

**Review by bazsi — APPROVED** [review]

**Pull request merged**